alternator/rjson.cc

```diff
@@ -113,6 +113,48 @@ void push_back(rjson::value& base_array, rjson::value&& item) {
 
 }
 
+bool single_value_comp::operator()(const rjson::value& r1, const rjson::value& r2) const {
+       auto r1_type = r1.GetType();
+       auto r2_type = r2.GetType();
+       switch (r1_type) {
+       case rjson::type::kNullType:
+           return r1_type < r2_type;
+       case rjson::type::kFalseType:
+           return r1_type < r2_type;
+       case rjson::type::kTrueType:
+           return r1_type < r2_type;
+       case rjson::type::kObjectType:
+           throw rjson::error("Object type comparison is not supported");
+       case rjson::type::kArrayType:
+           throw rjson::error("Array type comparison is not supported");
+       case rjson::type::kStringType: {
+           const size_t r1_len = r1.GetStringLength();
+           const size_t r2_len = r2.GetStringLength();
```

Comment:
But we just verified that r1 is a string - and here we also assume r2 is a string as well... How can we assume this?

Reply:
we can't, it's a bug (not quite reachable now since we currently use this comparison for instances of the same type only, but it's nonetheless a bug)